matching/views.py

```diff
@@ -47,14 +47,21 @@ def show_potential_matches(request, utrip_id):
     except Exception as e:
         print(e)
 
-    # filter out the current user from match pool
+    # users which have already being sent a matching request by current user, but not yet accepted
     current_matching_users = set(
         UserTripMatches.objects.filter(
             sender=current_user, match_status=MatchStatusEnum.PENDING.value
         ).values_list("receiver", flat=True)
     )
 
-    # Setting the Send Request button off for those users
+    # users who have sent a matching request to current user, pending/accepted as a match
+    matching_users = UserTripMatches.objects.filter(
+        receiver=current_user,
+        match_status__in=[MatchStatusEnum.PENDING.value, MatchStatusEnum.MATCHED.value],
+    ).values_list("sender", flat=True)
```

Comment:
These users are needed to be removed from match pool, to avoid circular dependency among sender and receiver. I've also changed the message in case of no potential users being displayed in `Show Potential Matches`